apps/web/src/query/resource-query/reference-model/types/reference-type.ts

```diff
@@ -0,0 +1,26 @@
+export interface ReferenceItem<Data = Record<string, any>> {
+    key: string;
+    label: string;
+    name: string;
+    color?: string;
+    icon?: string;
+    provider?: string;
+    continent?: {
+        continent_code?: string;
+        continent_label?: string;
+        latitude?: number;
+        longitude?: number;
+    };
+    latitude?: string;
+    longitude?: string;
+    data: Data;
+    description?: string;
+    link?: string;
+}
```

Comment:
`ReferenceItem`에 모든 속성이 들어있는데, 관심사에 따라 여러 개의 작은 인터페이스로 분리한 다음 마지막에 합치는 방식도 고려해볼 수 있을 것 같습니다. 머 이런?

Since all properties are currently defined in the `ReferenceItem` interface, you might consider splitting them into smaller interfaces based on concerns and then composing them back into ReferenceItem.

```typescript
interface ReferenceRegion {
    continent?: {
        continent_code?: string;
        continent_label?: string;
        latitude?: number;
        longitude?: number;
    };
    latitude?: string;
    longitude?: string;
}

export interface ReferenceItem extends ReferenceRegion {}
```

Reply:
These fields are for `region`? 

Then,, do you know why these fields are not designed to be included within `data` property?

Reply:
No

Reply:
But I really don't know...